src/Microsoft.TemplateEngine.Abstractions/GlobalSettings/IGlobalSettings.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.TemplateEngine.Abstractions.GlobalSettings
+{
+    /// <summary>
+    /// Interface that represents loading/storing data into settings.json file.
+    /// That is shared between multiple different hosts of TemplateEngine.
+    /// </summary>
+    public interface IGlobalSettings
+    {
+        /// <summary>
+        /// Triggered every time when settings change.
+        /// </summary>
+        event Action SettingsChanged;
+
+        /// <summary>
+        /// Returns uncached list of installer template packages.
+        /// </summary>
+        Task<IReadOnlyList<TemplatePackageData>> GetInstalledTemplatesPackagesAsync(CancellationToken cancellationToken);
+
+        /// <summary>
+        /// Stores list of installer template packages.
+        /// </summary>
+        Task SetInstalledTemplatesPackagesAsync(IReadOnlyList<TemplatePackageData> packages, CancellationToken cancellationToken);
+
+        /// <summary>
+        /// This method must be called before making any modifications to settings to ensure other processes on system
+        /// don't override or lose changes done by this process.
```

Comment:
I think in theory it is possible, but no likely. In this case pattern should be:
Call LockAsync()
Call SetInstalledTemplatesPackagesAsync()
Dispose the result of LockAsync.

Reply:
So if the pattern is lock / call set / unlock, would it make sense for the implementation to take the lock internally instead of putting the burden on the caller?

Reply:
Idea behind putting burden of lock on caller is that lock can last much longer than just get/set, idea is its locked for whole period of checking for installed packages and other checks and then installing... To ensure checks don't get outdated...
As Vlada mentioned before all this is now private so I'm resolving this converstation.